### DISCUSSION TITLE
How smart is the automatic TypeScript based prop generation supposed to be?

### DISCUSSION THREAD

**Discussion by johtso** [fwouts/previewjs#24] [Q&A] [answered]
I have a component that takes a relatively complex, but clearly defined input type:

```ts
type SeatPlanType = {
    id: SeatType['id'];
    status: SeatStatus;
}[][][]
```

reactpreview just generates the component with an empty array. Should I be expecting more?

**Comment by fwouts** [comment]
That's a good question. This probably works as intended. React Preview simply generates empty arrays by default. We could change that to generate one or more items instead. What do you think?

**Comment by fwouts** [comment, marked as answer]
Hi @johtso, I thought I'd let you know that React Preview now generates non-empty arrays since v0.4.3 :)
